packages/lesswrong/lib/sql/Query.ts

```diff
@@ -10,10 +10,17 @@ class Arg {
   public typehint = "";
 
   constructor(public value: any, type?: Type) {
+    if (this.value === null && type instanceof DefaultValueType && type.isNotNull() && type.getDefaultValueString()) {
+      if (type.isArray() || type.toConcrete() instanceof JsonType) {
+        this.value = type.getDefaultValue();
+      } else {
+        this.value = type.getDefaultValueString();
+      }
+    }
     // JSON arrays make node-postgres fall over, but we can work around it
     // with a special-case typehint
-    if (Array.isArray(value) && value[0] && typeof value[0] === "object") {
-      if (type instanceof JsonType) {
+    if (Array.isArray(this.value) && this.value[0] && typeof this.value[0] === "object") {
+      if (type?.toConcrete() instanceof JsonType) {
```

Comment:
This is the same change as in the nullability PR, just pulled forward: https://github.com/ForumMagnum/ForumMagnum/pull/8051/files#diff-76d49ec35292b020f26ecb5ad71bd0127acc27ec448a967cf6b41ae99f850253